config/initializers/0_ldap_authenticatable.rb

```diff
@@ -0,0 +1,109 @@
+require 'net/ldap'
+require 'devise/strategies/authenticatable'
+
+module Devise
+  module Strategies
+    class LdapAuthenticatable < Authenticatable
+      def valid?
+        Rails.configuration.x.use_ldap && params[:user].present?
+      end
+
+      def authenticate!
+        config = Rails.configuration
```

Comment:
devise.rb uses it to decide which key to use (ldap_username or email). I need to figure out how to make it not sensible to ordering though